dataland-frontend/src/components/resources/newLandingPage/AuthSection.vue

```diff
@@ -44,6 +75,38 @@ const isUserLoggedIn = ref<undefined | boolean>(undefined);
 const { isLandingPage } = defineProps<{
   isLandingPage: boolean;
 }>();
+const isDropdownOpen = ref(false);
+const isSmallScreen = ref(window.innerWidth < 992);
+
+const toggleDropdown = (): void => {
+  isDropdownOpen.value = !isDropdownOpen.value;
+};
+const handleResize = (): void => {
+  isSmallScreen.value = window.innerWidth < 992;
+};
+
+// Ensure to close the dropdown on clicking outside
+const handleClickOutside = (event: MouseEvent): void => {
+  const target = event.target as HTMLElement;
+  if (!target.closest('.dropdown')) {
+    isDropdownOpen.value = false;
+  }
+};
+
+onMounted(() => {
+  window.addEventListener('click', handleClickOutside);
```

Comment:
I'm not sure whether it is the best idea to use plain old javascript to listen for such events here; usually in our code, as far as I can see, we only handle resize events.